### ISSUE TITLE
SequenceTokenSliceLayer<T> missing deserialization constructor — blocks Clone() / DeepCopy() in any architecture using it

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1234]
## Summary

`SequenceTokenSliceLayer<T>` does not have a deserialization-friendly constructor, so any `NeuralNetworkBase<T>` that contains one cannot be cloned. This crashes `OptimizerBase.InitializeRandomSolution()` (which calls `Clone()` → `DeepCopy()` → `DeserializeInternalUnchecked()`), preventing every supervised optimizer from running on a `TransformerArchitecture<T>` that uses sequence slicing.

## Stack trace

Reproduced on AiDotNet `0.173.0` (with AiDotNet.Tensors `0.70.0`):

```
System.NotSupportedException : Layer type SequenceTokenSliceLayer`1 is not supported for deserialization (no known constructor found).
   at AiDotNet.Helpers.DeserializationHelper.CreateLayerFromType[T](String layerType, Int32[] inputShape, Int32[] outputShape, Dictionary`2 additionalParams)
   at AiDotNet.NeuralNetworks.NeuralNetworkBase`1.DeserializeInternalUnchecked(Byte[] data)
   at AiDotNet.NeuralNetworks.NeuralNetworkBase`1.DeepCopy()
   at AiDotNet.NeuralNetworks.NeuralNetworkBase`1.Clone()
   at AiDotNet.Optimizers.OptimizerBase`3.InitializeRandomSolution(TInput trainingData)
   at AiDotNet.Optimizers.AdamOptimizer`3.Optimize(OptimizationInputData`3 inputData)
   at AiDotNet.AiModelBuilder`3.BuildSupervisedInternalAsync(TInput x, TOutput y)
   at AiDotNet.AiModelBuilder`3.BuildAsync(CancellationToken cancellationToken)
```

## Root cause

`DeserializationHelper.CreateLayerFromType<T>` resolves layer types via reflection and looks for a constructor matching the deserialization signature (`(int[] inputShape, int[] outputShape, Dictionary<string, object> additionalParams)` or equivalent). Most layers in `AiDotNet.NeuralNetworks.Layers` provide this constructor; `SequenceTokenSliceLayer<T>` does not.

`OptimizerBase.InitializeRandomSolution` re-clones the architecture per random restart, so the bug is hit on the first `Optimize()` call regardless of whether the user has serialized anything.

## Repro

Build any `TransformerArchitecture<float>` that contains a sequence-slice layer (e.g., a transformer-LM that emits the last-token logits) and feed it to `AiModelBuilder<TInput, TOutput, TMetric>` with the default Adam optimizer:

```csharp
var arch = new TransformerArchitecture<float>(
    inputType: NeuralNetworkInputType.OneDimensional,
    taskType: NeuralNetworkTaskType.MultiClassClassification,
    inputDim: ctxLen,
    outputDim: vocabSize,
    dModel: 64,
    dFf: 256,
    numHeads: 4,
    numEncoderLayers: 1,
    numDecoderLayers: 0);

var builder = new AiModelBuilder<Matrix<float>, Vector<float>, MeanMetric>()
    .WithArchitecture(arch)
    .WithOptimizer(new AdamOptimizer<Matrix<float>, Vector<float>, MeanMetric>());

await builder.BuildAsync();  // <-- crashes here
```

## Impact

This blocks all classical-Transformer head-to-head baselines in HarmonicEngine that use the `AiModelBuilder` facade (the canonical AiDotNet supervised path). We have a workaround (a hand-rolled `MinimalTransformerPredictor`) for single-layer L4 head-to-heads, but the facade path is preferred for multi-layer (L=1/2/4/8) baselines because it picks the right scalar/tensor backends.

Specifically, this blocks the L4 self-attention layer's depth-sweep classical baseline at `tests/Learning/Phase_L4_AssociativeRetrieval_DepthSweep_Screen.cs` in the HarmonicEngine repo (PR #105).

## Suggested fix

Add the standard deserialization constructor to `SequenceTokenSliceLayer<T>`:

```csharp
public SequenceTokenSliceLayer(int[] inputShape, int[] outputShape, Dictionary<string, object> additionalParams)
    : base(inputShape, outputShape)
{
    // pull slice index / window from additionalParams
}
```

and ensure `Serialize()` writes those parameters into `additionalParams`. Any layer in `AiDotNet.NeuralNetworks.Layers` that already round-trips through `DeepCopy()` is a working template (e.g., `DenseLayer<T>`, `SoftmaxLayer<T>`).

## Environment

- AiDotNet 0.173.0
- AiDotNet.Tensors 0.70.0
- AiDotNet.Native.OpenBLAS 0.70.0
- .NET 10.0
- Windows 11 Pro 10.0.26220

## Related

- HarmonicEngine PR #105 (L4 self-attention) — depth-sweep classical baseline blocked
- Companion to ooples/AiDotNet#1232 (Transformer training convergence) and ooples/AiDotNet.Tensors#285 (OpenCL buffer cap) — both fixed in 0.173.0; this is the next blocker on the path to a fully working facade.
